runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -3947,6 +3947,48 @@ J9::CodeGenerator::collectSymRefs(
    return true;
    }
 
+/**
+ * \brief
+ *      This query is used by both fixUpProfiledInterfaceGuardTest (a codegen level optimization) and virtual guard evaluators
+ *      to decide whether a NOP guard should be generated. It's used on all platforms and compilation phases so that the decision
+ *      of generating VG NOPs is made in a consistent way.
+ *
+ * \param node
+ *      the virtual guard node
+ *
+ * \return
+ *      true if a NOP virtual guard should be generated. Otherwise, false.
+*/
+bool
```

Comment:
Can we place the documentation at the API specification level (in the header)?

Reply:
Documentation is added to the header.

Reply:
Moved it to the header.